### PR TITLE
all: set TLS 1.2 as the minimum required version

### PR DESCRIPTION
**Pull request by dominic**

### CODE REVIEW

File: core/core_test.go [hunk 1]
```diff
@@ -57,8 +57,8 @@ func TestForwardToLeader(t *testing.T) {
 		httpClient: &http.Client{
 			Transport: &http.Transport{
 				TLSClientConfig: &tls.Config{
-					RootCAs: certpool,
-				},
+					RootCAs:    certpool,
```
Comment (jeffomatic) [inline core/core_test.go]: Is this necessary for the tests to pass? Would we expect all clients to set this value in order to interact with Chain Core?

Comment (dominic) [inline core/core_test.go]: Per offline conversation, this is a simulated Chain Core API http object, not an external client object connecting to a Chain Core.

File: net/tls.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+func DefaultTLSConfig() *tls.Config {
+	return &tls.Config{
+		// Avoids most of the memorably-named TLS attacks
+		MinVersion: tls.VersionTLS12,
+		// Causes servers to use Go's default ciphersuite preferences,
+		// which are tuned to avoid attacks. Does nothing on clients.
+		PreferServerCipherSuites: true,
+		// Only use curves which have constant-time implementations
+		CurvePreferences: []tls.CurveID{
+			tls.CurveP256,
```
Comment (tarcieri) [inline net/tls.go]: I would suggest adding `tls.X25519` here, before `tls.CurveP256` (this was specifically recommended by @gtank despite what's on cryptopasta. Also Curve25519 rocks)

Comment (dominic) [inline net/tls.go]: Oops sorry! I totally added that, and then replaced it with a copypasta from his `main`. Adding it now.

**Pull request merged**